crates/aranya-daemon/src/daemon.rs

```diff
@@ -259,26 +259,36 @@ impl Daemon {
     }
 
     /// The daemon's entrypoint.
-    pub fn spawn(mut self) -> DaemonHandle {
+    pub async fn spawn(mut self) -> Result<DaemonHandle, ready::WaitError> {
         let _guard = self.span.enter();
         let mut set = JoinSet::new();
+        let waiter = ready::Waiter::new(3);
         set.spawn(
             self.sync_server
-                .serve()
+                .serve(waiter.notifier())
                 .instrument(info_span!("sync-server")),
         );
         set.spawn(
-            async move {
-                loop {
-                    if let Err(err) = self.syncer.next().await {
-                        error!(?err, "unable to sync with peer");
+            {
+                let notifier = waiter.notifier();
+                async move {
+                    notifier.notify();
```

Comment:
Does the `async move` introduce enough of a delay to be significant?
```suggestion
                waiter.notifier().notify();
                async move {
```

Reply:
The async block is the task, if I move it above then it will just immediately notify. The current version notifies once the task has been initially run once, which admittedly doesn't do much for us.

Reply:
Do we really need the notification for this particular task? I thought the notification is for tasks that have a delay before they are ready. I suppose there could be a delay before the task is scheduled to run for the first time.
It's mostly the QUIC server and API startup we are interested in. Even if the syncer task took a while to startup, the worst thing that happens is it takes a little longer to initiate a sync.

Reply:
Not really, I just figured it was nicer to do for all and then we wouldn't forget to add it if we add some setup for the syncer.